text/0088-metric-instrument-optional-refinements.md

```diff
@@ -0,0 +1,483 @@
+| Observer | sum-only |                 | non-negative  | non-negative-rate | _DeltaObserver_ |
+| Observer | sum-only | precomputed-sum |               | non-negative-rate | _CumulativeObserver_ |
+| Observer | sum-only |                 |               |                   | _UpDownDeltaObserver_ |
+| Observer | sum-only | precomputed-sum |               |                   | _UpDownCumulativeObserver_ |
+| Observer |          |                 | non-negative  |                   | _AbsoluteObserver_ |
+| Observer |          |                 |               |                   | _NonAbsoluteObserver_ |
+
+To arrive at this listing, several assumptions have been made.  For
+example, the precomputed-sum and non-negative-rate refeinments are
+only applicable in conjunction with a sum-only refinement.
+
+For the precomputed-sum instruments, we technically do not care
+whether the inputs are non-negative, because rate aggregation computes
+differences.  However, it is useful for other aggregations to assume
+that precomputed sums start at zero, and we will ignore the case where
+a precomputed sum has an initial value other than zero.
+
+#### Gauge instrument
+
+A Measure instrument with a default Last Value aggregation could be
+defined, hypothetically named a _Gauge_ instrument.  This would offer
+convenience for users that want this behavior, for there is otherwise
+no standard Measure refinement with Last Value aggregation.
+
+Sum-only uses for this hypothetical instrument should instead use
+either _CumulativeCounter_ or _UpDownCumulativeCounter_, since they
+are reporting a sum.  This (hypothetical) _Gauge_ instrument would be
+useful when a value is time-dependent and the average value is not of
+interest.
+
+## Internal details
+
+This is a change of understanding.  It does not request any new
+instruments be created or APIs be changed, but it does specify how we
+should think about adding new instruments.
+
+No API changes are called for in this proposal.
+
+### Translation into well-known systems
+
+#### Prometheus
+
+The Prometheus system defines four kinds of [synchronous metric
+instrument](https://prometheus.io/docs/concepts/metric_types).  
+
+| System     | Metric Kind  | Operation           | Aggregation          | Notes               |
+| ---------- | ------------ | ------------------- | -------------------- | ------------------- |
+| Prometheus | Counter      | Inc()               | Sum                  | Sum of positive deltas |
+| Prometheus | Counter      | Add()               | Sum                  | Sum of positive deltas |
+| Prometheus | Gauge        | Set()               | Last Value           | Non-additive or monotonic cumulative |
+| Prometheus | Gauge        | Inc()/Dec()         | Sum                  | Sum of deltas |
+| Prometheus | Gauge        | Add()/Sub()         | Sum                  | Sum of deltas |
+| Prometheus | Histogram    | Observe()           | Histogram            | |
```

Comment:
Histogram in Prometheus is restricted to only non-negative values :(

Reply:
OK! :)